src/plugins/WebOfTrust/util/SynchronousDelayedBackgroundJob.java

```diff
@@ -0,0 +1,231 @@
+     */
+    private void startNewJobThread() {
+        Thread jobThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                synchronized(SynchronousDelayedBackgroundJob.this) {
+                    long delay;
+                    // 1. Wait for the deadline to expire
+                    while ((delay = nextRunDeadline - System.currentTimeMillis()) > 0) {
+                        try {
+                            SynchronousDelayedBackgroundJob.this.wait(delay);
+                        } catch (InterruptedException e) {
+                            // Ignore. We can't really return here, since there is at
+                            // least one thread waiting for this thread to complete
+                            // (possibly including side effects caused by its run).
+                        }
+                    }
+                    // 2. Wait until the previous job is done, if any
+                    while (runningJobThread != null) {
+                        try {
+                            SynchronousDelayedBackgroundJob.this.wait();
+                        } catch (InterruptedException e) {
+                            // Ignore. This thread is not even visible from the
+                            // outside at this point, and we will be notified once the
+                            // currently running job is finished. We can't really return
+                            // here, since there is at least one thread waiting for this
+                            // thread to complete (possibly including side effects
+                            // caused by its run).
+                        }
+                    }
+                    // 3. Reset the nextRun variables
+                    nextRunDeadline = NO_DEADLINE;
+                    nextRunScheduled = false;
+                    // From here on, this thread is visible from the outside world,
+                    // and interruption should be handled correctly.
+                    runningJobThread = Thread.currentThread();
+                    // We cannot return here if we are terminating, since that would violate the
+                    // contract specified by {@link #triggerExecution()}. Instead, interrupt the
+                    // thread so that the job may terminate early.
+                    if (isTerminating) {
+                        runningJobThread.interrupt();
+                    }
+                }
+                try {
+                    // 4. Run the background job
+                    job.run();
+                } finally {
+                    synchronized(SynchronousDelayedBackgroundJob.this) {
+                        // 5. Increment runCount
+                        runCount++;
+                        runningJobThread = null;
+                        // 6. Notify all threads waiting upon completion
+                        SynchronousDelayedBackgroundJob.this.notifyAll();
```

Comment:
Now I think of it, we already have such a flag: `nextRunDeadline == NO_DEADLINE`. Using that.